server/api.go

```diff
@@ -167,6 +185,28 @@ func (a *apiServer) Publish(ctx context.Context, req *client.PublishRequest) (
 	}
 	a.logger.Debugf("api: Publish [subject=%s]", subject)
 
+	stream := a.metadata.GetStream(req.Stream)
+	if stream == nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("No such stream: %s", req.Stream))
+	}
+	partition, ok := stream.partitions[0] // Only check the first partition for now
+	if !ok {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("No partition in stream: %s", req.Stream))
+	}
+	if partition.paused {
+		// This partition (and thus, this stream) is paused.
+		// We need to re-create the stream before being able to use it.
+		protoPart := partition.Partition
+		for i := 0; i < len(stream.partitions); i++ {
```

Comment:
See discussion on https://github.com/liftbridge-io/liftbridge-api/pull/18 about pausing/resuming at the partition level.

Reply:
Implemented in a01f499.